docs/design/fused_moe_modular_kernel.md

```diff
@@ -0,0 +1,226 @@
+# Fused MoE Modular Kernel
+
+## Introduction
+FusedMoEModularKernel is implemented [here](https://github.com/vllm-project/vllm/blob/main/vllm/model_executor/layers/fused_moe/modular_kernel.py)
```

Comment:
If you use this syntax you'll see a nice GitHub icon next to the link in the docs

```suggestion
FusedMoEModularKernel is implemented [here](gh-file:vllm/model_executor/layers/fused_moe/modular_kernel.py)
```